models/track_and_reid_model.py

```diff
@@ -409,13 +420,19 @@ def create_or_load_tracklets(args, create_tracklets:bool):
     return tracklets
 
 
-def is_unknown_id(final_scores:dict, threshold:float=0.5):
+def is_unknown_id(reid_scores:dict, face_scores:dict, face_confs:np.array):
     unknown_id = False
-    ranks = sorted(final_scores.values(), reverse=True)
-    max_score = max(final_scores.values())
-    print(max_score)
-    if (ranks[0] - ranks[1]) <= 0.01 or threshold <= 0.4:
+    face_cos_sim = max(face_scores.values())
+    # an already labeled unknown reid already exists in the gallery and face_cos_sim is low
+    if ID_TO_NAME[max(reid_scores, key=reid_scores.get)] == 'Unknown' and face_cos_sim <= 0.4:
```

Comment:
You switch between face_cos_sim of 0.4 and 0.3 in different places, why is that?
Also, isn't this check redundant? If the image was labeled as 'Unknown' it means that the cos_sim was below a certain threshold, so why check it here again? Or is this the reason you use once 0.4 and once 0.3?